src/graph/FindPathExecutor.h

```diff
@@ -87,21 +98,46 @@ class FindPathExecutor final : public TraverseExecutor {
         std::multimap<VertexID, Path> &pathToNeighbor,
         VisitedBy visitedBy);
 
-    inline bool isPathAcceptable(Path path, Neighbor &neighbor, VisitedBy visitedBy);
+    inline bool isPathAcceptable(Path path, VertexID stepOutSrcId, Neighbor &neighbor,
+                                 VisitedBy visitedBy, bool dstShortest);
 
     Status setupVids();
 
     Status setupVidsFromRef(Clause::Vertices &vertices);
 
     Status doFilter(
         storage::StorageRpcResponse<storage::cpp2::QueryResponse> &&result,
-        Expression *filter,
-        bool isOutBound,
-        Frontiers &frontiers);
+        Frontiers &frontiers,
+        bool isToSide);
 
-    StatusOr<std::vector<storage::cpp2::PropDef>> getStepOutProps(bool reversely);
+    StatusOr<std::vector<storage::cpp2::PropDef>> getStepOutProps(bool isToSide);
 
-    StatusOr<std::vector<storage::cpp2::PropDef>> getDstProps();
+    StatusOr<std::vector<storage::cpp2::PropDef>> getNeighborProps(bool isToSide);
+
+    void fetchVertexPropsAndFilter(
+        storage::StorageRpcResponse<storage::cpp2::QueryResponse> &&result,
+        std::vector<VertexID> ids,
+        bool isToSide);
+
+    void doWithFilter(
+        storage::StorageRpcResponse<storage::cpp2::QueryResponse> &&result,
+        bool isToSide);
+    /**
+     * A container to hold the mapping from vertex id to its properties.
+     */
+    class VertexHolder final {
+    public:
+        explicit VertexHolder(ExecutionContext* ectx) : ectx_(ectx) { }
+        OptVariantType getDefaultProp(TagID tid, const std::string &prop) const;
+        OptVariantType get(VertexID id, TagID tid, const std::string &prop) const;
+        void add(const std::vector<storage::cpp2::QueryResponse> &responses);
+
+    private:
+        std::unordered_map<std::pair<VertexID, TagID>, RowReader> data_;
+        mutable std::unordered_map<
+            TagID, std::shared_ptr<const meta::SchemaProviderIf>> tagSchemaMap_;
+        ExecutionContext* ectx_{nullptr};
+    };
```

Comment:
It seems the `VertexHolder` has no difference with the one in `GoExecutor`, could we put the `VertexHolder` to `TraverseExecutor`?